Add a file to describe the welcome message components (#7)
rl-institut/open_plan: docs/view_components/welcome.rst

************
Welcome page
************

> After analyzing the post-Q3 meeting inputs regarding the tool interface design, all but one design have a welcome message as part of the

Reviewer: No need to mention Q3 meeting, this is meant to be the documentation for future users of the interface

linking ************
Welcome page
************

> A welcome window can be seen in almost every software/website in production. The welcome window should serve as an orientation point for the user and let them know, in as brief words (and other audiovisual content) as possible, what to expect from the tool. It is also the launch point from where the user begins their energy systems simulation.

Reviewer: The first sentence is a general statement which does not bring much to the description

Author: Removed now

The welcome page is to be implemented as an optional pop-up that springs up on the landing page when the user launches the tool. In this section, a structure for the welcome page is proposed, subject to further discussion with the stakeholders.

Attributes
----------

> The following are the view-components available to the user in the welcome pop-up on the landing window immediately after launching the software for the first time:

Reviewer: The view-component is the welcome view. I agree that a label or a link to a video are also "view-components" in some sense, but there are very generic or elementary. We want to describe here the "view-components" of our GUI which is a bit more specific. I would propose that we keep the term view-component for elements of our GUI

Author: I propose that we separate the term view-component into View and Components (following the way the wiki page for GUI describes it [here](https://en.wikipedia.org/wiki/List_of_graphical_user_interface_elements). Right now it is confusing. 

Window = components + the way they are rendered (here, Welcome pop-up which is nothing but a window)
Components = all the elements in the view such as pop-ups, links, menus, icons, buttons, etc. 
Rendering = how these things are rendered (the welcome view is rendered as a pop-up on the landing page)

#. **Welcome message on the pop-up**

>     * A text message 'Welcome to open_plan, the free and open-source energy systems planning and simulation tool based on the *OEMOF* framework and *energy cells* concept

Reviewer: The description should just say " A welcome text", not necessarily providing the text here. The text's content will be thought though in a second step: if we decide not to have a Welcome window after consultation, you effort to find a text are then in vain.

Author: Since I already wrote it, I will add 'A welcome text along the lines of' to the sentence and leave it as it is.

#. **Video intro message**

>     * A short, 3 minute video clip that provides the important information such as tool features, capabilities, where to find help, etc., very briefly

Reviewer: This description of attribute goes more into the direction of a description of a view component. We are thinking about the structure of the GUI here, not its content yet. Thinking the content now is pointless because the structure might change after discussions, we need to agree to have a video first before we think about what will be on the video (I repeat myself purposefully here to make sure I make my point clear :))

Author: So, just mentioning the components of the welcome window are enough then right? 

But since I already wrote them, I will leave them as they are for now.

.. _welcome-label:

Welcome page
------------

The welcome window serves as an orientation point for the user and let them know, in as brief words (and other audiovisual content) as possible, what to expect from the tool. It is also the launch point from where the user begins their energy systems simulation.

> The welcome page is to be implemented as an optional pop-up that springs up on the landing page when the user launches the tool. In this section, a structure for the welcome page is proposed, subject to further discussion with the stakeholders.

Reviewer: ```suggestion

```
The information contained here is redundant with the one you list below in action, we don't need it

Attributes
^^^^^^^^^^

> #. **Welcome message on the pop-up**

Reviewer: ```suggestion
#. **Welcome message**
```

>     * A single line text message

Reviewer: ```suggestion
    * A text
```

#. **Video intro message**
    * A short, 3 minute video clip that provides the important information such as tool features, capabilities, where to find help, etc., very briefly
    * This clip is embedded into the welcome pop-up. The user can view it on YouTube (or Vimeo) on a separate tab of the browser by clicking on a link on the video

#. **Check-box to disable the welcome pop-up**

#. **Hyperlinks**
    * Clickable links to read more, documentation, examples, use-cases, video tutorials, code-base (GitHub repo) and contact us

Actions
^^^^^^^

> * Pop-up that springs up everytime the software is launched freshly

Reviewer: ```suggestion
* Pop-up that springs up everytime the software is started
```

This is more a requirement

.. _welcome-label:

Welcome page
------------

> The welcome window serves as an orientation point for the user and let them know, in as brief words (and other audiovisual content) as possible, what to expect from the tool. It is also the launch point from where the user begins their energy systems simulation.

Reviewer: If this *is* the launch point, this implies we can only launch it from here, which is not necessarily true

Author: you are right. The landing page view is the launching point and the welcome pop-up is only one of the view-components on this landing page. I will remove the last sentence.

Attributes
^^^^^^^^^^

#. **Welcome message**
    * A text

#. **Video intro message**
    * A short, 3 minute video clip that provides the important information such as tool features, capabilities, where to find help, etc., very briefly
    * This clip is embedded into the welcome pop-up. The user can view it on YouTube (or Vimeo) on a separate tab of the browser by clicking on a link on the video

#. **Check-box to disable the welcome pop-up**

#. **Hyperlinks**
    * Clickable links to read more, documentation, examples, use-cases, video tutorials, code-base (GitHub repo) and contact us

Actions
^^^^^^^

> * Pop-up that springs up everytime the software is started

Reviewer: I think this belong to the requirements, because the user cannot act on the fact that the welcome view-component will be visible the first time they open the tool

* Checking the tick-box would disable the pop-up from appearing everytime the software is launched

> * The user could later re-enable the appearance of the pop-up window through *Display* or *Preferences* menu in *Settings*

Reviewer: Re-enabling is also a requirement as it cannot be performed on the welcome-view, but it will be via the file menu most likely

Author: changed

* Clicking on the hyperlinks will open the respective topic in a separate tab on the browser (e.g.: GitHub repo)

Requirement
^^^^^^^^^^^

* If the user ticks the check-box *Do not show again*, the pop-up should not appear during the subsequent launch of the tool

Link with other view-components
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> * Only connected to the landing view, which appears everytime the tool is launched

Reviewer: I would say it is linked to the file menu view component (from there you can make the welcome view-component appear again)

Author: will add your suggestion to the links section

.. _welcome-label:

Welcome page
------------

The welcome window serves as an orientation point or introduction for the user and let them know, in as brief words (and other audiovisual content) as possible, what to expect from the tool.

Attributes
^^^^^^^^^^

* **Welcome message**
    * A text

* **Video intro message**
    * A short, 3 minute video clip that provides the important information such as tool features, capabilities, where to find help, etc., very briefly
    * This clip is embedded into the welcome pop-up. The user can view it on YouTube (or Vimeo) on a separate tab of the browser by clicking on a link on the video

> * **Check-box to disable the welcome pop-up**
> 

Reviewer: ```suggestion
* **Check-box `Do not show again`**
    Disable/enable the display of the welcome view-component when tool is started.

```

* **Hyperlinks**
    * Clickable links to read more, documentation, examples, use-cases, video tutorials, code-base (GitHub repo) and contact us

Actions
^^^^^^^

> * Checking the tick-box would disable the pop-up from appearing everytime the software is launched

Reviewer: ```suggestion
* Checking the `Do not show again` check-box would disable the pop-up from appearing everytime the software is launched
```

* Clicking on the hyperlinks will open the respective topic in a separate tab on the browser (e.g.: GitHub repo)

Requirement
^^^^^^^^^^^

> * Pop-up that springs up everytime the software is started

Reviewer: ```suggestion
* By default, the welcome view-component should be displayed to the user everytime the software is started
```

> * If the user ticks the check-box *Do not show again*, the pop-up should not appear during the subsequent launch of the tool

Reviewer: ```suggestion
* If the user ticks the check-box *Do not show again*, the welcome view-component should not appear to the user during any subsequent launch of the tool
```

* The user could later re-enable the appearance of the pop-up window through *Display* or *Preferences* menu in *Settings*

Link with other view-components
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> * Connected to the landing view, which appears everytime the tool is launched

Reviewer: ```suggestion
ref:`landing-view`:

```

> * Connected also to the file menu; user can re-enable appearance of the welcome view-component through options within the file menu

Reviewer: ```suggestion
* ref:`file menu`:
     user can re-enable appearance of the welcome view-component through options within the file menu
```

Rendering of the view-component
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> 

Reviewer: ```suggestion
The welcome view component could be either a popup on top of the ref`:landing-view`, a part of the landing view or a view on its own.
```

Welcome page
************

The welcome window serves as an orientation point for the user and let them know, in as brief words (and other audiovisual content) as possible, what to expect from the tool. It is also the launch point from where the user begins their energy systems simulation.

The welcome page is to be implemented as an optional pop-up that springs up on the landing page when the user launches the tool. In this section, a structure for the welcome page is proposed, subject to further discussion with the stakeholders.

Attributes
----------

> The following are the components available to the user in the welcome pop-up on the landing window immediately after launching the software for the first time:

Reviewer: We actually don't need an introductory text for what attributes are, this belongs to the concept definition

#. **Welcome message on the pop-up**

>     * A text message along the lines of 'Welcome to open_plan, the free and open-source energy systems planning and simulation tool based on the *OEMOF* framework and *energy cells* concept

Reviewer: Simply "welcome message"
